src/start/syntax.md

```diff
@@ -1 +1,438 @@
+A lambda expression is of the form: `fn`, pattern, `=>`, expression.
+The lambda expression itself is a value — a value of function type[.](../../img/cheerio.jpg)
+The `=>` in lambda expressions correspond to the `->` in their types.
+The `->` arrows are right-associative infix type constructors denoting function types.
+Apply lambda expressions via prefix application — before the immediate operand.
+
+|               Expression                |            Evaluates To           |        Type        |
+|:---------------------------------------:|:---------------------------------:|:------------------:|
+| `(fn (x : int,y : int) => x + y)`       | `(fn (x : int,y : int) => x + y)` | `int * int -> int` |
+| `(3,4)`                                 | `(3,4)`                           | `int * int`        |
+| `(fn (x : int,y : int) => x + y) (3,4)` | `7`                               | `int`              |
+
+#### Function Binding
+
+Using a lambda expression more than once requires retyping it. We can give it a name instead.
+The `val` and `fun` keywords bind a lambda expression to an identifier, creating a *named function*.
+Take note that the `=` for binding differs from the `=>` reserved word.
+```sml
+(* add : int * int -> int *)
+val add : int * int -> int = fn (x,y) => x + y
+
+(* add : int * int -> int *)
+fun add (x : int,y : int) : int = x + y
+```
+
+Both function bindings for `add` above have the same value:
+`(fn (x,y) => x + y) : int * int -> int`.
+
+A named function can be thought of as a lambda expression that has been "identified".
+The bindings to `add` identify (or name) an otherwise anonymous function.
+Its value is the lambda expression it binds.
+
+|          Expression         |         Value         |        Type        |
+|:---------------------------:|:---------------------:|:------------------:|
+| `add`                       | `(fn (x,y) => x + y)` | `int * int -> int` |
+| `(fn (x,y) => x + y)`       | `(fn (x,y) => x + y)` | `int * int -> int` |
+| `add (3,4)`                 | `7`                   | `int`              |
+| `(fn (x,y) => x + y) (3,4)` | `7`                   | `int`              |
+
+#### Patterns and Case Expressions
+
+Patterns are present in every lambda expression, `case` expression, `val`, and `fun` binding.
+Every `fn` clause, `case` clause, and `fun` clause contains a pattern with a corresponding expression.
+A clause is of the form: pattern, `=>`, expression. Clauses are delimited by pipes `|`.
+
+|                     Expression                     | Example Clause   |  Clause Pattern  | Clause Expression |
+|:--------------------------------------------------:|:----------------:|:----------------:|:-----------------:|
+| `(fn (x,y) => x + y)`                              | `(x,y) => x + y` | `(x,y)`          | `x + y`           |
+| `(fn true => 1 | false => 0)`                      | `true => 1`      | `true`           | `1`               |
```

Comment:
you may need to escape these pipes - it looks like (when i built it) that it got interpreted as the end of the column

Reply:
Done.